test/unit/math/gpu/cholesky_decompose_test.cpp

```diff
@@ -0,0 +1,81 @@
+#ifdef STAN_OPENCL
+#include <stan/math/prim/mat.hpp>
+#include <stan/math/gpu/copy.hpp>
+#include <stan/math/gpu/cholesky_decompose.hpp>
+#include <stan/math/prim/mat/fun/cholesky_decompose.hpp>
+#include <stan/math/prim/mat/fun/Eigen.hpp>
+#include <stan/math/prim/mat/err/check_pos_definite.hpp>
+#include <stan/math/prim/mat/err/check_square.hpp>
+#include <stan/math/prim/mat/err/check_symmetric.hpp>
+#include <gtest/gtest.h>
+#include <algorithm>
+#define EXPECT_MATRIX_NEAR(A, B, DELTA) \
+  for (int i = 0; i < A.size(); i++)    \
+    EXPECT_NEAR(A(i), B(i), DELTA);
+
+TEST(MathMatrix, cholesky_decompose_cpu_vs_gpu_small) {
+  stan::math::matrix_d m0(3, 3);
+  m0 << 25, 15, -5, 15, 18, 0, -5, 0, 11;
+
+  stan::math::matrix_d m1(4, 4);
+  m1 << 18, 22, 54, 42, 22, 70, 86, 62, 54, 86, 174, 134, 42, 62, 134, 106;
+
+  stan::math::matrix_gpu m0_gpu(m0);
+  stan::math::matrix_gpu m1_gpu(m1);
+
+  stan::math::matrix_d m0_res = stan::math::cholesky_decompose(m0);
+  stan::math::matrix_d m1_res = stan::math::cholesky_decompose(m1);
+
+  stan::math::matrix_gpu m0_chol_gpu
+      = stan::math::cholesky_decompose(m0_gpu, floor(m0.rows() * 0.05));
+  stan::math::matrix_gpu m1_chol_gpu
+      = stan::math::cholesky_decompose(m1_gpu, floor(m0.rows() * 0.05));
```

Comment:
Can we add a few tests for simple and extreme values of `min_block`? Like is the answer still correct if you pass 1 and a few numbers that do not divide evenly into the size of the matrix, and couple numbers higher than the number of rows in the matrix?

Reply:
I have this test ready, will add them.

Reply:
Sean in my last commit I made a wrapper so that min block is not user facing. Do we still need these tests?

Otherwise should the function throw errors for min_block sizes that are silly like 1 or Inf?